port_ocean/run.py

```diff
@@ -45,6 +47,11 @@ def run(
         "app", default_app
     )
 
+    # Validate that the current integration's runtime matches the execution parameters
+    asyncio.get_event_loop().run_until_complete(
+        validate_integration_runtime(app.port_client, app.config.runtime)
+    )
```

Comment:
It means the application already filled in the defaults 
You need to catch that in the integration initialization